Tests fail if there is no aport for the curent device

I don't exactly know how to put it... But look at the logs.
I last worked on the device "sony-seagull", then switched branch to work on something else and run the tests. pmbootstrap remembered my choice, but it seems that it is confused by the non-existent device information.

Proposed solution: store the suggestions, run the tests and restore them. There might be another, easier way.

```
test/test_build_is_necessary.py:34: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pmb/parse/arguments.py:388: in arguments
    setattr(args, "deviceinfo", pmb.parse.deviceinfo(args))
_ _ _ _ _ _ _ _ _ ___ _ _ _ _ _ _ _ _ _ _ _ _ 

args = Namespace(action='chroot', alpine_version='edge', aports='/run/media/mayeul/DATA/postmarketos/pmbootstrap/aports', arc...rope/Paris', ui='weston', user='pmuser', verbose=False, work='/run/media/mayeul/DATA/postmarketos/pmbootstrap_workdir')
device = 'sony-seagull'

    def deviceinfo(args, device=None):
        """
        :param device: defaults to args.device
        """
        if not device:
            device = args.device
    
        if not os.path.exists(args.aports):
            logging.fatal("Aports directory is missing")
            logging.fatal("Please provide a path to the aports directory using the -p flag")
            raise RuntimeError("Aports directory missing")
    
        aport = args.aports + "/device/device-" + device
        if not os.path.exists(aport) or not os.path.exists(aport + "/deviceinfo"):
            logging.fatal("You will need to create a device-specific package")
            logging.fatal("before you can continue. Please create at least the")
            logging.fatal("following files:")
            logging.fatal(aport + "/APKBUILD")
            logging.fatal(aport + "/deviceinfo")
>           raise RuntimeError("Incomplete device information")
E           RuntimeError: Incomplete device information

pmb/parse/deviceinfo.py:43: RuntimeError
------------ Captured stderr setup ----------------
deviceinfo.py               38 CRITICAL You will need to create a device-specific package
deviceinfo.py               39 CRITICAL before you can continue. Please create at least the
deviceinfo.py               40 CRITICAL following files:
deviceinfo.py               41 CRITICAL /run/media/mayeul/DATA/postmarketos/pmbootstrap/aports/device/device-sony-seagull/APKBUILD
deviceinfo.py               42 CRITICAL /run/media/mayeul/DATA/postmarketos/pmbootstrap/aports/device/device-sony-seagull/deviceinfo
--------------------- Captured log setup -------------------
deviceinfo.py               38 CRITICAL You will need to create a device-specific package
deviceinfo.py               39 CRITICAL before you can continue. Please create at least the
deviceinfo.py               40 CRITICAL following files:
deviceinfo.py               41 CRITICAL /run/media/mayeul/DATA/postmarketos/pmbootstrap/aports/device/device-sony-seagull/APKBUILD
deviceinfo.py               42 CRITICAL /run/media/mayeul/DATA/postmarketos/pmbootstrap/aports/device/device-sony-seagull/deviceinfo
=================== 1 passed, 13 error in 2.28 seconds==================
```

--- a/test/testcases_fast.sh
+++ b/test/testcases_fast.sh
@@ -14,7 +14,7 @@ disabled="
 cd "$(dirname "$0")/.."
 device="$(./pmbootstrap.py config device)"
 deviceinfo="$PWD/aports/device/device-$device/deviceinfo"
-if ! [ -e "./aports/device/device-$device/deviceinfo" ]; then
+if ! [ -e "$deviceinfo" ]; then
 	echo "ERROR: Could not find deviceinfo file for selected device '$device'."
 	echo "Expected path: $deviceinfo"
 	echo "Maybe you have switched to a branch where your device does not exist?"

--- a/test/testcases_fast.sh
+++ b/test/testcases_fast.sh
@@ -10,9 +10,20 @@ disabled="
 	version
 "
 
+# Make sure we have a valid device (#1128)
+cd "$(dirname "$0")/.."
+device="$(./pmbootstrap.py config device)"
+deviceinfo="$PWD/aports/device/device-$device/deviceinfo"
+if ! [ -e "./aports/device/device-$device/deviceinfo" ]; then
+	echo "ERROR: Could not find deviceinfo file for selected device '$device'."
+	echo "Expected path: $deviceinfo"
+	echo "Maybe you have switched to a branch where your device does not exist?"
+	echo "Use 'pmbootstrap config device qemu-amd64' to switch to a valid device."
+	exit 1
+fi
+
 # Filter out disabled testcases
 enabled=""
-cd "$(dirname "$0")/.."
 for file in test/test_*.py; do
 	for test in $disabled; do
 		[ "test/test_${test}.py" = "$file" ] && continue 2

